rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/230_composite.yml

```diff
@@ -1278,3 +1278,155 @@ setup:
+
+---
+"Simple Composite aggregation with missing_bucket":
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test
+        body:
+          aggregations:
+            test:
+              composite:
+                sources: [
+                    "kw": {
+                      "terms": {
+                        "field": "keyword",
+                        "missing_bucket": true
+                      }
+                    }
+                ]
+
+  - length: { aggregations.test.buckets: 3 }
+  - match: { aggregations.test.buckets.0.key.kw: null }
+  - match: { aggregations.test.buckets.0.doc_count: 2 }
+
+---
+"Simple Composite aggregation with missing_order":
+  - skip:
+      # TODO: replace with 7.15.99 once PR is backported to 7.16
+      version: " - 7.99.0"
+      reason: "`missing_order` has been introduced in 7.16"
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test
+        body:
+          aggregations:
+            test:
+              composite:
+                sources: [
+                  "kw": {
+                    "terms": {
+                      "field": "keyword",
+                      "missing_bucket": true,
+                      "missing_order": "last"
+                    }
+                  }
+                ]
+
+  - length: { aggregations.test.buckets: 3 }
+  - match: { aggregations.test.buckets.2.key.kw: null }
+  - match: { aggregations.test.buckets.2.doc_count: 2 }
+
+---
+"missing_order with missing_bucket = false":
+  - skip:
+      # TODO: replace with 7.15.99 once PR is backported to 7.16
+      version: " - 7.99.0"
+      reason: "`missing_order` has been introduced in 7.16"
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test
+        body:
+          aggregations:
+            test:
+              composite:
+                sources: [
+                    "kw": {
+                      "terms": {
+                        "field": "keyword",
+                        "missing_bucket": false,
+                        "missing_order": "first"
+                      }
+                    }
+                ]
+
+  - length: { aggregations.test.buckets: 2 }
+  - match: { aggregations.test.buckets.0.key.kw: "bar" }
+  - match: { aggregations.test.buckets.1.key.kw: "foo" }
+
+---
+"missing_order without missing_bucket":
+  - skip:
+      # TODO: replace with 7.15.99 once PR is backported to 7.16
+      version: " - 7.99.0"
+      reason: "`missing_order` has been introduced in 7.16"
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test
+        body:
+          aggregations:
+            test:
+              composite:
+                sources: [
+                    "kw": {
+                      "terms": {
+                        "field": "keyword",
+                        "missing_order": "first" # ignored because missing_bucket is not true
```

Comment:
You're probably right, throwing an error in this case should help the user to make sure that they also set `missing_bucket` to true. After all, this differs from the behavior of `"missing": "_last"` in the search API where you don't have to set another parameter which might come as a surprise.

My thinking was along the lines that it might be useful to have this parameter ignored if you have code where you want a constant `missing_order` but toggle `missing_bucket` like
```
def agg(boolean includeMissing) {
   ...
   compSource.missingBucket(includeMissing).missingOrder("first")
   ...
}
```
This becomes slightly less convenient if missing_order is forbidden with `missing_bucket = false`.

But since it's easier to go from more restrictive to less restrictive and users have to get to know how to use the feature correctly, I'll change it to throw an error 👍